api/v2_vulnerabilities.go

```diff
@@ -310,9 +316,112 @@ type VulnerabilityHost struct {
 	} `json:"machineTags"`
-	Mid       int       `json:"mid"`
-	Severity  string    `json:"severity"`
-	StartTime time.Time `json:"startTime"`
-	Status    string    `json:"status"`
-	VulnID    string    `json:"vulnId"`
+	Props     VulnerabilityHostProps `json:"props"`
+	Mid       int                    `json:"mid"`
+	Severity  string                 `json:"severity"`
+	StartTime time.Time              `json:"startTime"`
+	Status    string                 `json:"status"`
+	VulnID    string                 `json:"vulnId"`
+}
+
+func (v *VulnerabilityHost) PackageActive() string {
+	if v.FeatureKey.PackageActive == 0 {
+		return ""
+	}
+	return "ACTIVE"
+}
+
+func (v *VulnerabilityHost) CvssV2() (cvssV2Score string) {
+	if v.CveProps.Metadata != nil {
+		score := v.CveProps.Metadata.NVD.CVSSv2.Score
+		cvssV2Score = strconv.FormatFloat(score, 'f', 1, 64)
+	}
+	return
+}
+
+func (v *VulnerabilityHost) CvssV3() (cvssV3Score string) {
+	if v.CveProps.Metadata != nil {
+		score := v.CveProps.Metadata.NVD.CVSSv3.Score
+		cvssV3Score = strconv.FormatFloat(score, 'f', 1, 64)
+	}
+	return
+}
+
+type VulnerabilityHostMetadata struct {
+	NVD struct {
+		CVSSv2 struct {
+			PublishedDateTime string  `json:"PublishedDateTime"`
+			Score             float64 `json:"Score"`
+			Vectors           string  `json:"Vectors"`
+		} `json:"CVSSv2"`
+		CVSSv3 struct {
+			ExploitabilityScore float64 `json:"ExploitabilityScore"`
+			ImpactScore         float64 `json:"ImpactScore"`
+			Score               float64 `json:"Score"`
+			Vectors             string  `json:"Vectors"`
+		} `json:"CVSSv3"`
+	} `json:"NVD"`
+}
+
+type VulnerabilityHostProps struct {
+	FirstTimeSeen   *time.Time `json:"first_time_seen,omitempty"`
+	IsDailyJob      int        `json:"isDailyJob,omitempty"`
+	LastUpdatedTime *time.Time `json:"last_updated_time,omitempty"`
+}
+
+func (v *VulnerabilityHost) HasFix() bool {
+	return v.FixInfo.FixAvailable == "1"
+}
+
+func (hosts *VulnerabilitiesHostResponse) VulnerabilityCounts() HostVulnCounts {
+	var hostCounts = HostVulnCounts{}
+
+	// remove duplicates before count.
+	for _, h := range hosts.Data {
+		switch h.Severity {
+		case "Critical":
+			hostCounts.Critical++
+			hostCounts.Total++
+			if h.HasFix() {
+				hostCounts.CritFixable++
+				hostCounts.TotalFixable++
+			}
+		case "High":
+			hostCounts.High++
+			hostCounts.Total++
+			if h.HasFix() {
+				hostCounts.HighFixable++
+				hostCounts.TotalFixable++
+			}
+		case "Medium":
+			hostCounts.Medium++
+			hostCounts.Total++
+			if h.HasFix() {
+				hostCounts.MedFixable++
+				hostCounts.TotalFixable++
+			}
+		case "Low":
+			hostCounts.Low++
+			hostCounts.Total++
+			if h.HasFix() {
+				hostCounts.LowFixable++
+				hostCounts.TotalFixable++
+			}
+		default:
+			hostCounts.Info++
+			hostCounts.Total++
+			if h.HasFix() {
+				hostCounts.InfoFixable++
+				hostCounts.TotalFixable++
+			}
+		}
+	}
+
+	return hostCounts
+}
+
+type VulnCveSummary struct {
```

Comment:
This struct is massively used but it not being test it anywhere, I run some commands and it doesn't even have JSON tags so this shows as:
```
  "CVE-2022-39842": {
    "Count": 1,
    "Host": { ... },
    "Hostnames": [
      "ip-10-0-1-48"
    ]
  }
```

Reply:
Could you point to where this is being used in the json output? It should not be used in a json output.

Reply:
This is not used in the `api` package at all, move it to the `cli` since you are using it there.